dumpling/export/dump.go

```diff
@@ -81,11 +81,8 @@ func NewDumper(ctx context.Context, conf *Config) (*Dumper, error) {
 		selectTiDBTableRegionFunc: selectTiDBTableRegion,
 	}
 
-	if conf.PromFactory == nil {
-		d.metrics = defaultMetrics
-	} else {
-		d.metrics = newMetrics(conf.PromFactory, []string{})
-	}
+	d.metrics = newMetrics(conf.PromFactory, conf.Labels)
+	d.metrics.registerTo(conf.PromRegistry)
```

Comment:
I think we don't need this call and `conf.PromRegistry`. 
For Factory injected by dataflow engine, the Factory will automatically register the metrics variables when NewXXX in `newMetrics`.  So for other cases we can also imitate this behaviour. 
- For standalone dumpling binary, we let the default Factory become something like NewAutoRegisterFactory(NewDefaultFactory, ...)
where NewAutoRegisterFactory is at https://github.com/pingcap/tiflow/blob/329230aa9359836577f6a638168b4e717c045c24/engine/pkg/promutil/registry.go#L99

- And for DM's case DM should let the Factory argument has NewAutoRegisterFactory wrapped

Reply:
AutoRegisterFactory need a explicit Registry, that Registry can call `Unregister` when dumpling is about to close. https://github.com/pingcap/tiflow/blob/329230aa9359836577f6a638168b4e717c045c24/engine/pkg/promutil/registry.go#L79 In fact, offical prometheus also has a "AutoRegisterFactory" https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promauto , but compared to it, ours can handle Unregister.

in my understanding, the decision is more about other applications which want to use dumpling/lightning as a library? I now a bit agree that explitly requiring two things is less error-prone. Will review later.

also cc @maxshuang